lib/resolve.js

```diff
@@ -72,37 +101,62 @@ const singularize = function (matches) {
   }
 }
 
-exports.attachment = function (heroku, app, id, options = {}) {
-  const headers = options.headers || {}
+const appAttachment = function (heroku, app, id, options = {}) {
+  const headers = addActionsHeader(options.headers)
+  return heroku.post('/actions/addon-attachments/resolve', {
+    'headers': headers, 'body': {'app': app, 'addon_attachment': id, 'addon_service': options.addon_service || null}
+  }).then((attachments) => singularize(attachments))
```

Comment:
since singularize is just a normal function, you can just do `.then(singularize)`